plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/reflection/HtmlReflection.java

```diff
@@ -0,0 +1,199 @@
+     * ${@link HtmlReflection#findListOfElements(String, Class, Object)} for detailed
+     * description
+     *
+     * @param page the page on which the method will be executed
+     * @param listTitle value of ElementTitle annotation of required element
+     * @return list of WebElement's
+     * @throws PageException if nothing found or current page is not initialized
+     */
+    List<WebElement> findListOfElements(Page page, String listTitle) throws PageException;
+
+
+    /**
+     * Find elements list in context of required block See
+     * ${@link HtmlReflection#findListOfElements(String, Class, Object)} for detailed
+     * description
+     *
+     * @param page the page on which the method will be executed
+     * @param blockPath full path or just a name of the block to search
+     * @param listTitle value of ElementTitle annotation of required element
+     * @param type type of elements in list that is being searched for
+     * @param <T> type of elements in returned list
+     * @return list of elements of particular type
+     * @throws PageException if nothing found or current page is not initialized
+     */
+    <T extends WebElement> List<T> findListOfElementsInBlock(Page page, String blockPath, String listTitle, Class<T> type)
+            throws PageException;
+
+    /**
+     * Finds elements list in context of required block See
+     * ${@link HtmlReflection#findListOfElements(String, Class, Object)} for detailed
+     * description
+     *
+     * @param page the page on which the method will be executed
+     * @param blockPath full path or just a name of the block to search
+     * @param listTitle value of ElementTitle annotation of required element
+     * @return list of WebElement's
+     * @throws PageException if nothing found or current page is not initialized
+     */
+    List<WebElement> findListOfElementsInBlock(Page page, String blockPath, String listTitle) throws PageException;
+
+    /**
+     * See {@link HtmlReflection#findBlocks(String, Object, boolean)} for description.
+     * This wrapper finds only one block. Search is being performed on a current
+     * page
+     *
+     * @param page the page on which the method will be executed
+     * @param blockPath full path or just a name of the block to search
+     * @return first found block object
+     * @throws java.util.NoSuchElementException if couldn't find any block
+     */
+    HtmlElement findBlock(Page page, String blockPath) throws NoSuchElementException;
+
+    /**
+     * See {@link HtmlReflection#findBlocks(String, Object, boolean)} for description.
+     * Search is being performed on a current page
+     *
+     * @param page the page on which the method will be executed
+     * @param blockPath full path or just a name of the block to search
+     * @return list of objects that were found by specified path
+     */
+    List<HtmlElement> findBlocks(Page page, String blockPath) throws NoSuchElementException;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the declaration of thrown exception 'org.openqa.selenium.NoSuchElementException' which is a runtime exception. ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=101&issues=AWbDPCulSgN4uCclZPgX&open=AWbDPCulSgN4uCclZPgX)

[ISSUE_KEY]: AWbDPCulSgN4uCclZPgX